engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -472,18 +476,15 @@ class JoinPartitionKernel : public kernel {
 													right_join_types.cbegin(), right_join_types.cend());
 	}
 
-	static void partition_table(const std::string & kernel_id,
+	void partition_table(const std::string & kernel_id,
```

Comment:
This function is used for partitioning data for either left or right table. This means that depending on the table you are using the message tracker 0 or 1. And it also changes the destination table and the destination output cache. I don't think this is being done correctly please take another look

Reply:
You're right, it was inconsistent. I changed the logic.